nbuild/stdenv/install.py

```diff
@@ -21,43 +22,56 @@ def make_symlink(src, dst):
     os.symlink(src, dst)
 
 
-def make_cp(*src, dest, root='', args=''):
+def make_cp(*files, dest, root=''):
     package = current_build().current_package
-    for filename in src:
-        path = f'{package.install_dir}/{root}/{filename}/'
-        cmd(f'cp {args} {path} {dest}')
+    for filename in files:
+        path = f'{package.install_dir}/{root}/{filename}'
+        shutil.copy2(path, dest)
 
 
-def make_mkdir(dir, args=''):
+def make_mkdir(dir):
     package = current_build().current_package
     path = f'{package.install_dir}/{dir}/'
-    cmd(f'mkdir {args} {path}')
+    if not os.path.exists(path):
+        os.makedirs(path)
 
 
-def make_rm(*files, root='', args=''):
+def make_rm_file(*files, root=''):
     package = current_build().current_package
     for filename in files:
-        path = f'{package.install_dir}/{root}/{filename}/'
-        cmd(f'rm {args} {path}')
+        path = f'{package.install_dir}/{root}/{filename}'
+        os.remove(path)
 
 
-def make_mv(*src, dest, root='', args=''):
+def make_rm_directory(dir, root=''):
     package = current_build().current_package
-    for filename in src:
-        path = f'{package.install_dir}/{root}/{filename}/'
-        cmd(f'mv {args} {path} {dest}')
+    path = f'{package.install_dir}/{root}/{dir}'
+    os.rmdir(path)
+
+
+def make_mv(*files, dest, root=''):
+    package = current_build().current_package
+    for filename in files:
+        path = f'{package.install_dir}/{root}/{filename}'
+        shutil.move(path, dest)
 
 
-def make_chmod(dest, args):
+def make_chmod(dest, mode, root=''):
     package = current_build().current_package
-    path = f'{package.install_dir}/{dest}/'
-    cmd(f'chmod {args} {path}')
+    path = f'{package.install_dir}/{root}/{dest}'
+    octal_mode = "0o" + mode
+    os.chmod(path, octal_mode)
```

Comment:
`os.chmod` requires a number as its second argument, not a string.
You can make octal numbers in python by prefixing them with `0o`, so for example, `0o100` is `64`.
I think `make_chmod` should take a mode as an octal number as well, exactly as `os.chmod` does, and just forwards it to it.
```python
make_chmod('aaaaaaa', 0o755)
```
(which would call
```python
os.chmod(path, 0o755)
```
)